internal/buildengine/engine.go

```diff
@@ -813,9 +812,22 @@ func (e *Engine) build(ctx context.Context, moduleName string, builtModules map[
 		e.listener.OnBuildStarted(meta.module)
 	}
 
-	moduleSchema, deploy, err := build(ctx, meta.plugin, e.projectRoot, sch, meta.module.Config, e.buildEnv, e.devMode)
+	moduleSchema, deploy, err := build(ctx, meta.plugin, e.projectRoot, languageplugin.BuildContext{
+		Config:       meta.module.Config,
+		Schema:       sch,
+		Dependencies: meta.module.Dependencies,
+	}, e.buildEnv, e.devMode)
 	if err != nil {
 		terminal.UpdateModuleState(ctx, moduleName, terminal.BuildStateFailed)
+		if errors.Is(err, errInvalidateDependencies) {
+			go func() {
+				logger := log.FromContext(ctx)
```

Comment:
It's generally an anti-pattern to spawn a Goroutine like this. There's no concurrency limit, no way to stop it, no way to check its status. I would suggest having a think about a better approach here. Often it's preferable to use some combination of an errgroup, or a dedicated background worker attached to a channel, whereby the work is delivered on the channel.

Reply:
That makes sense. I'm going to remove it for now because it's hard to actually run this code without an external language plugin implementation